metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/resolver/BaseFunctionResolver.java

```diff
@@ -94,6 +95,16 @@ public void initialize(Context context) {
     this.context = context;
   }
 
+  /**
+   * Close the Stellar functions.
+   */
+  @Override
+  public void close() throws IOException {
+    for (StellarFunctionInfo info : getFunctionInfo()) {
```

Comment:
Should there be an "Attempt" to close each function?  I think we want to have a try catch for each call here maybe and just throw with the functions that failed afterwards.

Reply:
On a related note, we may want to consider timeouts at some point, but we don't do this for initialization either.

Reply:
What about something like this?
```
public void close() throws IOException {
    Map<String, Throwable> errors = new HashMap();
    for (StellarFunctionInfo info : getFunctionInfo()) {
      try {
        info.getFunction().close();
      } catch (Throwable t) {
        errors.put(info.getName(), t);
      }
    }
    if (!errors.isEmpty()) {
      StringBuilder sb = new StringBuilder();
      sb.append("Unable to close Stellar functions:");
      for (Map.Entry<String, Throwable> e : errors.entrySet()) {
        Throwable throwable = e.getValue();
        String eText = String
            .format("Exception - Function: %s; Message: %s; Cause: %s", e.getKey(), throwable .getMessage(),
                throwable .getCause());
        sb.append(System.lineSeparator());
        sb.append(eText);
      }
      throw new IOException(sb.toString());
    }
  }
```

Reply:
Yes, that is great.